src/test/java/org/jenkins_ci/plugins/flexible_publish/FlexiblePublisherTest.java

```diff
@@ -158,4 +171,322 @@ public void testMultipleConditionsMultipleActions() throws Exception {
+            
+            FreeStyleBuild b = p.scheduleBuild2(0).get();
+            assertBuildStatus(Result.FAILURE, b);
+            
+            // ArtifactArchiver is executed even prior publisher fails.
+            assertTrue(new File(b.getArtifactsDir(), "artifact.txt").exists());
+        }
+        
+        // Jenkins executes all publishers even one of them throws AbortException.
+        {
+            FreeStyleProject p = createFreeStyleProject();
+            
+            p.getBuildersList().add(new FileWriteBuilder("artifact.txt", "blahblahblah"));
+            p.getPublishersList().add(new ThorwAbortExceptionPublisher());
+            p.getPublishersList().add(new ArtifactArchiver("**/*", "", false));
+            
+            FreeStyleBuild b = p.scheduleBuild2(0).get();
+            assertBuildStatus(Result.FAILURE, b);
+            
+            // ArtifactArchiver is executed even prior publisher fails.
+            assertTrue(new File(b.getArtifactsDir(), "artifact.txt").exists());
+            
+            // Somehow Jenkins prints stacktrace for AbortException. You can see that here.
+            // System.out.println(b.getLog());
+        }
+        
+        // Jenkins executes all publishers even one of them throws any Exceptions.
+        {
+            FreeStyleProject p = createFreeStyleProject();
+            
+            p.getBuildersList().add(new FileWriteBuilder("artifact.txt", "blahblahblah"));
+            p.getPublishersList().add(new ThorwGeneralExceptionPublisher());
+            p.getPublishersList().add(new ArtifactArchiver("**/*", "", false));
+            
+            FreeStyleBuild b = p.scheduleBuild2(0).get();
+            assertBuildStatus(Result.FAILURE, b);
+            
+            // ArtifactArchiver is executed even prior publisher fails.
+            assertTrue(new File(b.getArtifactsDir(), "artifact.txt").exists());
+        }
+        
+        //// Flexible Publish should run as Jenkins core do.
+        
+        // Flexible Publish executes all publishers even one of them failed.
+        {
+            FreeStyleProject p = createFreeStyleProject();
+            
+            p.getBuildersList().add(new FileWriteBuilder("artifact.txt", "blahblahblah"));
+            p.getPublishersList().add(new FlexiblePublisher(Arrays.asList(
+                    new ConditionalPublisher(
+                            new AlwaysRun(),
+                            Arrays.<BuildStep>asList(
+                                    new FailurePublisher()
+                            ),
+                            new BuildStepRunner.Fail(),
+                            false,
+                            null,
+                            null
+                    ),
+                    new ConditionalPublisher(
+                            new AlwaysRun(),
+                            Arrays.<BuildStep>asList(
+                                    new ArtifactArchiver("**/*", "", false)

```

Comment:
Better create anonymous class with Publisher that will try to do NPE
```
String temp = null;
 temp.toString()
```
This will cause non AbortException error and exclude possible ArtifactArchiver hidden reactions.